js/modules/main/src/controllers/mjsController.js

```diff
@@ -40,6 +40,10 @@ var MjsController = function(mjs, notification, item, auth, $rootScope, $scope)
 	$rootScope.$on('name-updated', function() {
 		self.username = self.get_username(mjs.latest);
 	});
+
+	$rootScope.$on('item-removed', function() {
+		self.load(mjs.get_items_ids());
```

Comment:
Is the API get necessary? 
why not  get the item's string from the event and remove it from self.mjs_items yourself?